ArduPlane/soaring.cpp

```diff
@@ -19,48 +19,34 @@ void Plane::update_soaring() {
     
     g2.soaring_controller.update_vario();
 
-    // Check for throttle suppression change.
-    switch (control_mode->mode_number()) {
-    case Mode::Number::AUTO:
-    case Mode::Number::FLY_BY_WIRE_B:
-    case Mode::Number::CRUISE:
-        g2.soaring_controller.suppress_throttle();
-        break;
-    case Mode::Number::THERMAL:
-        // Never use throttle in THERMAL with soaring active.
-        g2.soaring_controller.set_throttle_suppressed(true);
-        break;
-    default:
-        // In any other mode allow throttle.
-        g2.soaring_controller.set_throttle_suppressed(false);
-        break;
-    }
+    if (control_mode->does_automatic_thermal_switch()) {
 
-    // Nothing to do if we are in powered flight
-    if (!g2.soaring_controller.get_throttle_suppressed() && aparm.throttle_max > 0) {
-        return;
-    }
+        // Check for throttle suppression change.
+        if (!g2.soaring_controller.suppress_throttle()) {
+            // Nothing to do if we are in powered flight
+            return;
```

Comment:
This return bugs me a bit... nothing else here is returning early, its all an elaborate `if` statement.  That would usually mean inverting the if clause and putting the code inside rather than outside...

Reply:
Switched it around, I agree it's cleaner.